core/src/main/scala/io/finch/internal/Accept.scala

```diff
@@ -0,0 +1,21 @@
+package io.finch.internal
+
+import com.twitter.util.Try
+import javax.activation.MimeType
+
+case class Accept(`type`: String, subtype: String) { self =>
+
+  def matches(other: Accept): Boolean = {
+    (self.`type` == other.`type` || self.`type` == "*") &&
+      (self.subtype == other.subtype || self.subtype == "*")
+  }
+
+}
+
+object Accept {
+
+  def apply(s: String): Option[Accept] = Try {
+    val mt = new MimeType(s)
```

Comment:
Is there a way to do that w/o relying on Java APIs?

Reply:
Also just using try-catch would probably be more efficient as it allows us to eliminate the `Try` allocation.

Reply:
Didn't want to use a regex because it could lead to potential ReDoS.
I could parse it manually though

https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html

Reply:
I see. Let use this Java thing for now and evolve it later if we want to.